src/auth/strategies/local.strategy.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common'
+import { PassportStrategy } from '@nestjs/passport'
+import { Strategy } from 'passport-local'
+
+import { ValidationError, NotFoundError } from '../../common/errors/errors'
+import { AuthService } from '../auth.service'
+import type { User } from '../../user/schemas/user.schema'
+
+@Injectable()
+export class LocalStrategy extends PassportStrategy(Strategy) {
+  constructor(private readonly authService: AuthService) {
+    super()
+  }
+
+  async validate(
+    username: string,
+    password: string,
+  ): Promise<Omit<User, 'password'>> {
+    try {
+      return await this.authService.getValidatedUser(username, password)
+    } catch (error) {
+      if (error instanceof NotFoundError) {
+        throw new NotFoundException(error.message)
+      }
+
+      if (error instanceof ValidationError) {
+        throw new BadRequestException(error.message)
+      }
+
+      throw new InternalServerErrorException(error.message)
```

Comment:
Two comments here:
1. `error` can be of any type, including primitives, and might not have `message` property. I suggest adding `"useUnknownInCatchVariables": true` to the `tsconfig.json` in order to highlight such misuse.

2. Error message can contain sensitive information that we're unlikely want to expose to the client, so for such a generic case (unconditional return of 500) it's better to not pass original message, instead returning something very generic (ie ”Internal server error‘).

Reply:
Fixed